src/CLI/clitree/cli-xml/mac.xml

```diff
@@ -0,0 +1,328 @@
+        </ACTION>
+    </COMMAND>
+
+
+<!--====================show mac address-table address <mac-address> command ======================-->
+    <COMMAND
+        help="show mac address table address mac-address"
+        name="show mac address-table address"
+        >
+        <PARAM
+            name="mac-addr"
+            help="48 bit MAC address"
+            ptype="MAC_ADDR"
+	    >
+	</PARAM>
+        <ACTION>
+		python $SONIC_CLI_ROOT/sonic-cli-mac.py get_openconfig_interfaces_interfaces show_mac.j2 mac-addr ${mac-addr}
+        </ACTION>
+    </COMMAND>
+
+<!--====================show mac address-table vlan <vlan-id> command ======================-->
+    <COMMAND
+        help="mac address-table for VLAN"
+        name="show mac address-table vlan"
+        >
+        <PARAM
+            name="vlan-id"
+            help="VLAN identifier (1-4094)"
+            ptype="VLAN_ID"
+	    >
+	</PARAM>
+        <ACTION>
+		python $SONIC_CLI_ROOT/sonic-cli-mac.py get_openconfig_interfaces_interfaces show_mac.j2 vlan ${vlan-id}
+        </ACTION>
+    </COMMAND>
+
+<!--====================show mac address-table interface <p-id> command ======================-->
+    <COMMAND
+        help="mac address-table for interfaces"
+        name="show mac address-table interface">
+	<PARAM
+	    name="if-subcommands"
+	    help="Interface subcommands"
+	    mode="switch"
+	    ptype="SUBCOMMAND"
+	    >
+	    <PARAM
+	        name="Ethernet"
+		help="Ethernet IEEE 802.3"
+		ptype="SUBCOMMAND"
+		mode="subcommand"
+		>
+		<PARAM
+		    name="phy-if-id"
+                    help="Physical interface(Multiples of 4)"
+		    ptype="UINT"
+		    >
+                </PARAM>
+            </PARAM>
+	    <PARAM
+	        name="Port-channel"
+		help="Port-channel ID (1-128)"
+		ptype="SUBCOMMAND"
+		mode="subcommand"
+		>
+		<PARAM
+		    name="po-id"
+                    help="port-channel id (1-128)"
+		    ptype="UINT"
+		    >
+                </PARAM>
+	    </PARAM>
+            </PARAM>
+        <ACTION>
+		if test "${if-subcommands}" = "Ethernet"; then&#xA;
+		    python $SONIC_CLI_ROOT/sonic-cli-mac.py get_openconfig_interfaces_interfaces show_mac.j2 interface Ethernet${phy-if-id}&#xA; 
+		elif test "${if-subcommands}" = "Port-channel"; then&#xA;
+		    python $SONIC_CLI_ROOT/sonic-cli-mac.py get_openconfig_interfaces_interfaces show_mac.j2 interface Port-channel${po-id}&#xA; 
+                else&#xA;
+		    python $SONIC_CLI_ROOT/sonic-cli-mac.py get_openconfig_interfaces_interfaces show_mac.j2 ${__full_line}&#xA;
+	        fi	
+        </ACTION>
+    </COMMAND>
+
+<!--====================show mac address-table static commands ======================-->
+    <COMMAND
+        help="mac address-table for static commands"
+        name="show mac address-table static">
+	<PARAM
+	    name="s-subcommands"
+	    help="static subcommands"
+	    mode="switch"
+	    ptype="SUBCOMMAND"
+	    optional="true"
```

Comment:
Do we need this optional flag?

Reply:
The requirement is to get result for "show mac address-table static" as well as for "show mac address-table static {vlan | interface | address}". Without optional param,  CLI will be complaining as an incomplete command and we won't get result for static.